river/tree/_attribute_observer/numeric_attribute_regression_quantizer_observer.py

```diff
@@ -0,0 +1,163 @@
+        self._x_var = Var()
+        self._quantizer = FeatureQuantizer(radius=self.radius)
+
+    def update(self, x, y, sample_weight):
+        if x is None:
+            return
+        else:
+            self._x_var.update(x, sample_weight)
+            self._quantizer.update(x, y, sample_weight)
+
+    def probability_of_attribute_value_given_class(self, x, y):
+        raise NotImplementedError
+
+    def best_evaluated_split_suggestion(
+        self, criterion, pre_split_dist, att_idx, binary_only=True
+    ):
+        candidate = AttributeSplitSuggestion(None, [{}], -math.inf)
+
+        p_x = None
+        for (x, left_dist) in self._quantizer:
+            # First hash element
+            if p_x is None:
+                # In case the hash carries just one element return the null split
+                if len(self._quantizer) == 1:
+                    return candidate
+                p_x = x
+                continue
+
+            right_dist = pre_split_dist - left_dist
+            post_split_dists = [left_dist, right_dist]
+            merit = criterion.merit_of_split(pre_split_dist, post_split_dists)
+
+            if merit > candidate.merit:
+                split_point = (p_x + x) / 2.0
+                candidate = self._update_candidate(
+                    split_point, att_idx, post_split_dists, merit
+                )
+
+            p_x = x
+        return candidate
+
+    @property
+    def x_var(self):
+        return self._x_var
+
+    @staticmethod
+    def _update_candidate(split_point, att_idx, post_split_dists, merit):
+        num_att_binary_test = NumericAttributeBinaryTest(att_idx, split_point, True)
+        candidate = AttributeSplitSuggestion(
+            num_att_binary_test, post_split_dists, merit
+        )
+
+        return candidate
+
+
+class Slot:
+    def __init__(
+        self, x: float, y=typing.Union[float, VectorDict], weight: float = 1.0
+    ):
+        self.x_stats = Mean()
+        self.x_stats.update(x, weight)
+
+        self.y_stats: typing.Union[Var, VectorDict]
+
+        self._update_estimator: typing.Callable[
+            [typing.Union[float, VectorDict], float], None
+        ]
+        self.is_single_target = True
+
+        self._init_estimator(y)
+        self._update_estimator(y, weight)
+
+    def _init_estimator(self, y):
+        if isinstance(y, dict):
+            self.is_single_target = False
+            self.y_stats = VectorDict(default_factory=functools.partial(Var))
+            self._update_estimator = self._update_estimator_multivariate
+        else:
+            self.y_stats = Var()
+            self._update_estimator = self._update_estimator_univariate
+
+    def _update_estimator_univariate(self, target, sample_weight):
+        self.y_stats.update(target, sample_weight)
+
+    def _update_estimator_multivariate(self, target, sample_weight):
+        for t in target:
+            self.y_stats[t].update(target[t], sample_weight)
+
+    def __iadd__(self, o):
```

Comment:
What does `o` mean? If observer then what about `obs` or `observer`?

Reply:
Nothing that fancy :D

I think pycharm did the trick. As far as I am concerned, `o` stands for "other". Should I change it?

Reply:
My personal preference is to avoid variable names with a single character as it can become somewhat obscure. If there is a meaningful word why not use it. Being said that, I am ok with using `o` in this case.